### ISSUE TITLE
Old jobs: Job logs not always available

### ISSUE THREAD

**Issue by eapearson** [kbase/execution_engine2#139]
Here is an example of a old job, from Friday, September 30, 2016 23:03:50:

```json
{
    "version": "1.1",
    "result": [
        {
            "user": "eaptest30",
            "authstrat": "DEFAULT",
            "wsid": -1,
            "status": "error",
            "updated": 1475276646979,
            "running": 1475276640790,
            "finished": 1475276646966,
            "errormsg": "/kb/module/bin/../lib/genome_transform/genome_transformServer.pm terminating\n",
            "msg": "/kb/module/bin/../lib/genome_transform/genome_transformServer.pm terminating\n Execution engine job for genome_transform.genbank_to_genome",
            "error_code": 0,
            "scheduler_type": "awe",
            "job_input": {
                "wsid": -1,
                "method": "genome_transform.genbank_to_genome",
                "requested_release": "dev",
                "params": [
                    {
                        "workspace": "eaptest30:1475276576555",
                        "contigset_id": "ath_hy5_demo2_contigset",
                        "genbank_file_path": "/data/bulktest/eaptest30//ath_hy5_demo2.fq copy",
                        "genome_id": "ath_hy5_demo2.fq-copy"
                    }
                ],
                "service_ver": "dde33725d2aa5f333169caccdabcfa8a227a7e53",
                "app_id": "genome_transform.genbank_to_genome",
                "source_ws_objects": [],
                "narrative_cell_info": {}
            },
            "job_id": "57eeef56e4b0b05cf8996c02",
            "created": 1475276630000
        }
    ],
    "id": "123"
}
```

This is what ee2 says now:

```json
{
  "version": "1.1",
  "id": "123",
  "method": "execution_engine2.get_job_logs",
  "params": [
    {
      "job_id": "57eeef56e4b0b05cf8996c02"
    }
  ]
}
```

```json
{
    "error": {
        "code": -32000,
        "name": "Server error",
        "message": "'Cannot find job log with id: 57eeef56e4b0b05cf8996c02'",
        "error": "Traceback (most recent call last):\n  File \"/kb/module/scripts/../lib/execution_engine2/execution_engine2Server.py\", line 101, in _call_method\n    result = method(ctx, *params)\n  File \"/kb/module/lib/execution_engine2/execution_engine2Impl.py\", line 341, in get_job_logs\n    job_id=params[\"job_id\"], skip_lines=params.get(\"skip_lines\", None)\n  File \"/kb/module/lib/execution_engine2/SDKMethodRunner.py\", line 225, in view_job_logs\n    return self._get_job_log(job_id, skip_lines)\n  File \"/kb/module/lib/execution_engine2/SDKMethodRunner.py\", line 197, in _get_job_log\n    log = self.get_mongo_util().get_job_log(job_id)\n  File \"/kb/module/lib/execution_engine2/db/MongoUtil.py\", line 162, in get_job_log\n    \"Cannot find job log with id: {}\".format(job_id)\nexecution_engine2.exceptions.RecordNotFoundException: Cannot find job log with id: 57eeef56e4b0b05cf8996c02\n"
    },
    "id": "123",
    "version": "1.1"
}
```

But njsw says this:

```json
{
    "id": "123",
    "version": "1.1",
    "method": "NarrativeJobService.get_job_logs",
    "params": [
        {
          "job_id": "57eeef56e4b0b05cf8996c02"
        }
    ]
}
```

```json
{
    "version": "1.1",
    "result": [
        {
            "lines": [
                {
                    "line": "Running on dtn (127.0.1.1), in /mnt/awe/ci/work/06/91/35/06913528-0fba-48f3-9409-8ee7b760ea6f_0_0",
                    "is_error": 0
                },
                {
                    "line": "Client group: kb_upload",
                    "is_error": 0
                },
                {
                    "line": "Server version of Execution Engine: 0.2.5 (matches to version of runner script)",
                    "is_error": 0
                },
                {
                    "line": "Image name received from catalog: dockerhub-ci.kbase.us/kbase:genome_transform.dde33725d2aa5f333169caccdabcfa8a227a7e53",
                    "is_error": 0
                },
                {
                    "line": "Job runner recieved callback URL: http://172.17.0.1:44455",
                    "is_error": 0
                },
                {
                    "line": "Use of uninitialized value in concatenation (.) or string at /kb/module/bin/../lib/genome_transform/genome_transformServer.pm line 279.",
                    "is_error": 0
                },
                {
                    "line": "/data/bulktest/eaptest30//ath_hy5_demo2.fq copy$VAR1 = {",
                    "is_error": 0
                },
                {
                    "line": "          'genome_id' => 'ath_hy5_demo2.fq-copy',",
                    "is_error": 0
                },
                {
                    "line": "          'contigset_id' => 'ath_hy5_demo2_contigset',",
                    "is_error": 0
                },
                {
                    "line": "          'genbank_file_path' => '/data/bulktest/eaptest30//ath_hy5_demo2.fq copy',",
                    "is_error": 0
                },
                {
                    "line": "          'workspace' => 'eaptest30:1475276576555'",
                    "is_error": 0
                },
                {
                    "line": "        };",
                    "is_error": 0
                },
                {
                    "line": "file-path  /data/bulktest/eaptest30//ath_hy5_demo2.fq copy",
                    "is_error": 0
                },
                {
                    "line": "",
                    "is_error": 0
                },
                {
                    "line": "mkdir /kb/module/work/tmp/Genomes",
                    "is_error": 0
                },
                {
                    "line": "creating a temp/expDir direcotory for data processing, continuing..",
                    "is_error": 0
                },
                {
                    "line": "file /data/bulktest/eaptest30//ath_hy5_demo2.fq copy mimetype is [text/plain]",
                    "is_error": 0
                },
                {
                    "line": "checking if \"/data/bulktest/eaptest30//ath_hy5_demo2.fq copy\", of \"text/plain\" is unsupported",
                    "is_error": 0
                },
                {
                    "line": "no action - return original filename",
                    "is_error": 0
                },
                {
                    "line": "relevant shock URL is [https://ci.kbase.us/services/shock-api]",
                    "is_error": 0
                },
                {
                    "line": "system_and_check [/kb/deployment/bin/trns_transform_Genbank_Genome_to_KBaseGenomes_Genome --shock_service_url https://ci.kbase.us/services/shock-api --workspace_service_url https://ci.kbase.us/services/ws --workspace_name eaptest30:1475276576555 --object_name ath_hy5_demo2.fq-copy --contigset_object_name ath_hy5_demo2_contigset --input_directory /data/bulktest/eaptest30//ath_hy5_demo2.fq copy --working_directory /kb/module/work/tmp/Genomes]",
                    "is_error": 0
                },
                {
                    "line": "2016-09-30 23:04:04,390 - trns_transform_Genbank_Genome_to_KBaseGenomes_Genome.py - 42 - INFO - Starting transformation of Genbank to KBaseGenomes.Genome",
                    "is_error": 0
                },
                {
                    "line": "indir /data/bulktest/eaptest30/ath_hy5_demo2.fq",
                    "is_error": 0
                },
                {
                    "line": "wsname eaptest30:1475276576555",
                    "is_error": 0
                },
                {
                    "line": "wsurl https://ci.kbase.us/services/ws",
                    "is_error": 0
                },
                {
                    "line": "isTest false",
                    "is_error": 0
                },
                {
                    "line": "workdir 1 /kb/module/work/tmp/Genomes",
                    "is_error": 0
                },
                {
                    "line": "input /data/bulktest/eaptest30/ath_hy5_demo2.fq",
                    "is_error": 0
                },
                {
                    "line": "java.io.FileNotFoundException: /data/bulktest/eaptest30/ath_hy5_demo2.fq (No such file or directory)",
                    "is_error": 0
                },
                {
                    "line": "\tat java.io.FileInputStream.open(Native Method)",
                    "is_error": 0
                },
                {
                    "line": "\tat java.io.FileInputStream.<init>(FileInputStream.java:146)",
                    "is_error": 0
                },
                {
                    "line": "\tat java.util.Scanner.<init>(Scanner.java:656)",
                    "is_error": 0
                },
                {
                    "line": "\tat us.kbase.genbank.ConvertGBK.splitRecord(ConvertGBK.java:253)",
                    "is_error": 0
                },
                {
                    "line": "\tat us.kbase.genbank.ConvertGBK.init(ConvertGBK.java:234)",
                    "is_error": 0
                },
                {
                    "line": "\tat us.kbase.genbank.ConvertGBK.<init>(ConvertGBK.java:79)",
                    "is_error": 0
                },
                {
                    "line": "\tat us.kbase.genbank.ConvertGBK.main(ConvertGBK.java:791)",
                    "is_error": 0
                },
                {
                    "line": "Tool execution returned exit status 12016-09-30 23:04:04,472 - trns_transform_Genbank_Genome_to_KBaseGenomes_Genome.py - 152 - ERROR - Transformation from Genbank.Genome to KBaseGenomes.Genome failed on /data/bulktest/eaptest30//ath_hy5_demo2.fq",
                    "is_error": 0
                },
                {
                    "line": "Traceback (most recent call last):",
                    "is_error": 0
                },
                {
                    "line": "  File \"/kb/deployment/pybin/trns_transform_Genbank_Genome_to_KBaseGenomes_Genome.py\", line 149, in <module>",
                    "is_error": 0
                },
                {
                    "line": "    working_directory=args.working_directory)",
                    "is_error": 0
                },
                {
                    "line": "  File \"/kb/deployment/pybin/trns_transform_Genbank_Genome_to_KBaseGenomes_Genome.py\", line 85, in transform",
                    "is_error": 0
                },
                {
                    "line": "    raise Exception(\"Transformation from Genbank.Genome to KBaseGenomes.Genome failed on {0}\".format(input_directory))",
                    "is_error": 0
                },
                {
                    "line": "Exception: Transformation from Genbank.Genome to KBaseGenomes.Genome failed on /data/bulktest/eaptest30//ath_hy5_demo2.fq",
                    "is_error": 0
                },
                {
                    "line": "2016-09-30 23:04:04,472 - trns_transform_Genbank_Genome_to_KBaseGenomes_Genome.py - 152 - ERROR - Transformation from Genbank.Genome to KBaseGenomes.Genome failed on /data/bulktest/eaptest30//ath_hy5_demo2.fq",
                    "is_error": 0
                },
                {
                    "line": "Traceback (most recent call last):",
                    "is_error": 0
                },
                {
                    "line": "  File \"/kb/deployment/pybin/trns_transform_Genbank_Genome_to_KBaseGenomes_Genome.py\", line 149, in <module>",
                    "is_error": 0
                },
                {
                    "line": "    working_directory=args.working_directory)",
                    "is_error": 0
                },
                {
                    "line": "  File \"/kb/deployment/pybin/trns_transform_Genbank_Genome_to_KBaseGenomes_Genome.py\", line 85, in transform",
                    "is_error": 0
                },
                {
                    "line": "    raise Exception(\"Transformation from Genbank.Genome to KBaseGenomes.Genome failed on {0}\".format(input_directory))",
                    "is_error": 0
                },
                {
                    "line": "Exception: Transformation from Genbank.Genome to KBaseGenomes.Genome failed on /data/bulktest/eaptest30//ath_hy5_demo2.fq",
                    "is_error": 0
                },
                {
                    "line": "system() cmd exited with value 1",
                    "is_error": 0
                },
                {
                    "line": "Error: JSONRPCError: /kb/module/bin/../lib/genome_transform/genome_transformServer.pm terminating\n\n/kb/module/bin/../lib/genome_transform/genome_transformServer.pm terminating\n",
                    "is_error": 1
                }
            ],
            "last_line_number": 52
        }
    ]
}
```

**Comment by eapearson** [comment]
Sorry, this is a duplicate of https://github.com/kbase/execution_engine2/issues/121, but has better documentation.

**Comment by bio-boris** [comment]
The job logs script probably failed on this record for some reason. It was failing for some records.  https://github.com/kbase/execution_engine2/blob/develop/lib/execution_engine2/db/models/transfer/transfer_logs.py

**Comment by bio-boris** [comment]
Added a fix for this in #81 
https://github.com/kbase/execution_engine2/blob/transfer_scripts1/lib/execution_engine2/db/models/transfer/transfer_logs.py#L79-L81

We just need to re-run the logs script and maybe keep logs for existing jobs and move them in or just purge them

**Comment by bio-boris** [comment]
Fixed as per #121
